packages/trpc/server/template-router/router.ts

```diff
@@ -41,6 +42,13 @@ export const templateRouter = router({
       try {
         const { templateId } = input;
 
+        const limits = await getServerLimits({ email: ctx.user.email });
+
+        if (limits.remaining.documents === 0) {
+          throw new Error('You have reached your document limit.');
+          return;
```

Comment:
I'm assuming the reason you did 

```
      const documentFromTemplate = await createDocumentFromTemplate({
        templateId,
        templateId,
      });
```

In the prior code is since it could be undefined?

The undefined is being caused by this redundant return statement because of the throw error.

If you remove this return, you can revert the old code back to use `{ id }`

Reply:
Ohh, I see.